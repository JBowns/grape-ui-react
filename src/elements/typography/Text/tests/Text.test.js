import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from '..';
import 'jest-styled-components';

describe('Text Component base', () => {
  it('should return a Text object with base styling', () => {
    const component = renderer.create(<Text>Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Text Component specified styles', () => {
  it('should return a Text object with a specified color', () => {
    const component = renderer.create(<Text color="green.light">Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Text object with specified font-family', () => {
    const component = renderer.create(<Text fontFamily="monospace">Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Text object with specified font-weight', () => {
    const component = renderer.create(<Text fontWeight="bolder">Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Text object with font-style: italic', () => {
    const component = renderer.create(<Text italic>Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Text object with specified kerning', () => {
    const component = renderer.create(<Text kerning="1px">Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Text object with font-size: lg', () => {
    const component = renderer.create(<Text lg>Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Text object with font-size: sm', () => {
    const component = renderer.create(<Text sm>Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Text object with a specified text-align', () => {
    const component = renderer.create(<Text textAlign="right">Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Text object with a specified text-decoration', () => {
    const component = renderer.create(<Text textDecoration="underline">Lorem Ipsum</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
