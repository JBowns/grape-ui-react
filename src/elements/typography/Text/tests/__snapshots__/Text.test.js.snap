// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component base should return a Text object with base styling 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<span
  className="c0"
>
  Lorem Ipsum
</span>
`;

exports[`Text Component specified styles should return a Text object with a specified color 1`] = `
.c0 {
  color: hsl(122.4,39.4%,69.2%);
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<span
  className="c0"
>
  Lorem Ipsum
</span>
`;

exports[`Text Component specified styles should return a Text object with a specified text-align 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: right;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<span
  className="c0"
>
  Lorem Ipsum
</span>
`;

exports[`Text Component specified styles should return a Text object with a specified text-decoration 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<span
  className="c0"
>
  Lorem Ipsum
</span>
`;

exports[`Text Component specified styles should return a Text object with font-size: lg 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 1.2rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<span
  className="c0"
>
  Lorem Ipsum
</span>
`;

exports[`Text Component specified styles should return a Text object with font-size: sm 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<span
  className="c0"
>
  Lorem Ipsum
</span>
`;

exports[`Text Component specified styles should return a Text object with font-style: italic 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-style: italic;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<span
  className="c0"
>
  Lorem Ipsum
</span>
`;

exports[`Text Component specified styles should return a Text object with specified font-family 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: Consolas,"Andale Mono WT","Andale Mono","Lucida Console","Lucida Sans Typewriter","DejaVu Sans Mono","Bitstream Vera Sans Mono","Liberation Mono","Nimbus Mono L",Monaco,"Courier New",Courier,monospace;
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<span
  className="c0"
>
  Lorem Ipsum
</span>
`;

exports[`Text Component specified styles should return a Text object with specified font-weight 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: bolder;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<span
  className="c0"
>
  Lorem Ipsum
</span>
`;

exports[`Text Component specified styles should return a Text object with specified kerning 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

<span
  className="c0"
>
  Lorem Ipsum
</span>
`;
